### PR TITLE
Fix code coverage at codecov to work

### PR DESCRIPTION
**Pull request by rchiodo**
Make code coverage work by actually building with source maps. This should make a PR generate code coverage too.

My last run shows this:
https://codecov.io/gh/microsoft/vscode-jupyter/tree/3bc258d841b809aa9bc3013fbd23684327ac3345

-   [ ] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [ ] Title summarizes what is changing.
-   [ ] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1974,6 +1974,7 @@
         "gulp-azure-storage": "^0.11.1",
         "gulp-filter": "^5.1.0",
         "gulp-rename": "^1.4.0",
+        "gulp-sourcemaps": "^3.0.0",
```
Comment (rchiodo) [inline package.json:1977]: Blech, don't need this.  Will remove

File: src/test/coverage.ts [hunk 2]
```diff
@@ -13,19 +13,21 @@ export function setupCoverage() {
     if (!process.env.VSC_JUPYTER_INSTRUMENT_CODE_FOR_COVERAGE) {
         return;
     }
-    const reports = ['lcovonly', 'text', 'text-summary', 'html'];
+    const reports = ['text', 'text-summary'];
     const NYC = require('nyc');
     const nyc = new NYC({
         cwd: path.join(EXTENSION_ROOT_DIR_FOR_TESTS),
-        include: ['**/out/client/**/*.js'],
+        extension: ['.ts'],
```
Comment (rchiodo) [inline src/test/coverage.ts:20]: This makes it use source maps

File: gulpfile.js [hunk 3]
```diff
@@ -21,18 +21,19 @@ const flat = require('flat');
...
-    tsProject
-        .src()
-        .pipe(tsProject())
-        .on('error', () => (failed = true))
-        .js.pipe(gulp.dest('out'))
-        .on('finish', () => (failed ? done(new Error('TypeScript compilation errors')) : done()));
+gulp.task('compile', async (done) => {
+    // Use tsc so we can generate source maps that look just like tsc does (gulp-sourcemap does not generate them the same way)
+    try {
+        const stdout = await spawnAsync('tsc', ['-p', './'], {}, true);
```
Comment (rchiodo) [inline gulpfile.js:29]: This is necessary so we end up with the same output we would if we built locally.

Comment (rchiodo) [reply gulpfile.js:29]: And the source maps are correct. Gulp-sourcemaps had a bunch of issues (couldn't rewrite sourceroot)

File: .github/workflows/insiders.yml [hunk 4]
```diff
@@ -216,11 +217,11 @@ jobs:
       # Upload unit test coverage reports for later use in the "reports" job.
       - name: Upload unit test coverage reports
         uses: actions/upload-artifact@v2
+        if: always()
```
Comment (IanMatthewHuff) [inline .github/workflows/insiders.yml:220]: Thanks, I was just about to move this myself when I saw the warnings.

**Review by IanMatthewHuff — APPROVED** [review]

**Pull request merged**